external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -2937,12 +2940,82 @@ fn module_call_impl(
     (call, return_)
 }
 
+/// If the constant that we are referencing has an `error` attribute, we need to change the type of
+/// the constant to a u64 since this will be compiled into a u64 error code.
+fn annotated_error_const(context: &mut Context, e: &mut T::Exp) {
+    if let sp!(
+        const_loc,
+        T::UnannotatedExp_::Constant(module_ident, constant_name)
+    ) = &mut e.exp
+    {
+        let ConstantInfo {
+            attributes,
+            defined_loc,
+            signature,
+        } = context.constant_info(module_ident, constant_name);
+        let has_error_annotation =
+            attributes.contains_key_(&known_attributes::ErrorAttribute.into());
+        let u64_type = matches!(
+            signature.value.builtin_name(),
+            Some(sp!(_, N::BuiltinTypeName_::U64))
+        );
+        let ty_loc = e.ty.loc;
+        let const_def_loc = *defined_loc;
+
+        if has_error_annotation
+            && context.env.check_feature(
+                context.current_package(),
+                FeatureGate::CleverAssertions,
+                *const_loc,
+            )
+        {
+            e.ty = N::Type_::builtin(ty_loc, sp(ty_loc, N::BuiltinTypeName_::U64), vec![]);
```

Comment:
This scares me a lot lol. Just having a constant with the wrong type. 
Maybe we should add a new AST node for errors? This could also help with my comment in `naming` w.r.t. swapping out the value early

Reply:
I get your concern, but honestly this doesn't really scare me since in particular, this _is_ the right type. Like it _will_ be and is (I would argue) a `u64`. I'm hesitant to add it to the AST as it feels a bit heavyweight to me...

Reply:
Yeah but you risk just panicking future type-aware passes like constant folding. Not a huge concern, but I think adding some resiliency here would be good, and address the naming issue too